logic.py

```diff
@@ -0,0 +1,47 @@
+"""
+This module has the logic to find all solutions to the queens problem
+"""
+from solution_builder import SolutionBuilder, Solution
+from copy import deepcopy
+
+
+def find_solutions(board_size: int) -> list:
+    """
+    Find all of the solution to the queens problem to a board with a given size
+    :param board_size: The size of the board
+    :return: All of the solution
+    """
+    builder = SolutionBuilder(board_size)
+    return build_solution(builder)
+
+
+def build_solution(builder: SolutionBuilder) -> list:
+    """
+    Build solutions recursively with a SolutionBuilder one row at a time
+    :param builder: the solution builder
+    :return: A list of all solutions possible with the prev steps
+    """
+    if builder.is_solution_completed():
+        return [builder.get_solution()]
+    free_spaces = builder.get_free_places_in_next_row()
+    solutions = []
+    for next_space in free_spaces:
+        new_builder = deepcopy(builder)
+        new_builder.place_queen(next_space)
+        solutions = union_solutions(solutions, build_solution(new_builder))
+    return solutions
+
+
+def union_solutions(solutions_list1: list, solutions_list2: list) -> list:
+    """
+    Union 2 lists of solutions.
+    Note that this function doesn't check for duplicates in the lists beforehand.
+    :param solutions_list1: A first list of solutions
+    :param solutions_list2:A second list of solutions
+    :return: A union of the 2 lists
+    """
+    union = solutions_list1[:]
```

Comment:
יכול להסתכם ב:
list(set(solutions_list1 + solutions_list2))
הלא כן? מרגיש לי שהפונקציה מיותרת.. מסכים איתי?

Reply:
I checked it  out and for an object to be in a set it needs to be hashable

Reply:
נכון, צודק, קיבלתי